decentraland/helpers/storage.js

```diff
@@ -7,47 +8,73 @@ const MY_ASSETS_LIST = "MY_ASSETS_LIST";
 
 export const storeEphemeralAccount = async account => {
   const { privateKey } = account;
-  await _storeData(EPHEMERAL_ACCOUNT_PRIV_KEY, privateKey);
+  await _storeData(EPHEMERAL_ACCOUNT_PRIV_KEY, privateKey, true);
 };
 
 export const retrieveEphemeralAccount = async () => {
   let account = null;
 
-  const privateKey = await _retrieveData(EPHEMERAL_ACCOUNT_PRIV_KEY);
+  const privateKey = await _retrieveData(EPHEMERAL_ACCOUNT_PRIV_KEY, true);
   if (privateKey != null) account = createFromPrivateKey(privateKey);
 
   return account;
 };
 
 export const storeMyAssets = async myAssets => {
-  const strMyAssets = JSON.stringify(myAssets);
-  await _storeData(MY_ASSETS_LIST, strMyAssets);
+  const strMyAssets = _fromJson(myAssets);
+  await _storeData(MY_ASSETS_LIST, strMyAssets, false);
 };
 
 export const retrieveMyAssets = async () => {
-  const strMyAssets = await _retrieveData(MY_ASSETS_LIST);
-  const myAssets = JSON.parse(strMyAssets);
+  const strMyAssets = await _retrieveData(MY_ASSETS_LIST, false);
+  const myAssets = _toJson(strMyAssets);
```

Comment:
Hmm, when we're retrieving them don't we want them to become an object rather than JSON?

Reply:
I think you have them flipped

Reply:
Hmmm, I'm meaning JSON = object and String = JSON object parsed to string.

Reply:
Correct me if I'm wrong but for me Json (JavaScript Object Notation) is the way that JS represents their objects.
So, I've meant
```javascript
json = toJson = { a: 1, b: 2, c: 3 }
string = fromJson = '{"a":1,"b":2,"c":3}'
```

Reply:
I think JS represents their objects in one way, and JSON is a _notation_ for representing those objects as strings in a way that as closely resembles the object representation as possible. So JavaScript objects !== JSON

Reply:
:thinking: 
Hmmm... I agree with your argument but it stills confusing for me since I always read `parse` functions as something like that: `Type.parse(data) => convert data to type`, so, using that logic 
`obj = JSON.parse(string) => convert string to JSON and assign to obj` and `obj === stringifyedObject === JSON`.
Anyway, could be better rename functions to `toString/fromString` or `toObject/fromObject` since that same misconception could occur with the code readers?

Reply:
> `obj = JSON.parse(string) => convert string to JSON and assign to obj` and `obj === stringifyedObject === JSON`.

This ^ is the part I don't think is correct. Here's my understanding (and I'm 99% sure of this):

`obj = JSON.parse(string)` => convert string to a JavaScript object, and the utility library that happens to handle this is named `JSON` (because of its ability to turn objects into JSON). Assign to `obj` and `obj !== stringifiedObject === JSON`.

Reply:
(Note that REST APIs use JSON, and that would only be possible with strings, not objects.)

Reply:
> Anyway, could be better rename functions to `toString/fromString` or `toObject/fromObject` since that same misconception could occur with the code readers?

This works for me!

Reply:
I think that I've found the missing 1% here :smile: :
`JSON is a text syntax that facilitates structured data interchange between all programming languages.` - 1st line of ECMA-404.pdf

Reply:
Hahaha

Reply:
Done, thanks for that discussion!

Reply:
Thanks to you too! Great, resolving this.